core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeInMemorySorter.java

```diff
@@ -118,26 +124,28 @@ public int compare(RecordPointerAndKeyPrefix r1, RecordPointerAndKeyPrefix r2) {
   public UnsafeInMemorySorter(
     final MemoryConsumer consumer,
     final TaskMemoryManager memoryManager,
-    final RecordComparator recordComparator,
+    final RecordComparator.Factory recordComparatorFactory,
```

Comment:
Why do we need to change `UnsafeInMemorySorter`. The `TaskContext` refers `UnsafeExternalSorter`, so we only need the comparator factory in `UnsafeExternalSorter`.